test/corpus/alter.txt

```diff
@@ -697,3 +697,292 @@ ALTER SCHEMA sales OWNER TO CURRENT_USER;
       (keyword_owner)
       (keyword_to)
       (identifier))))
+
+================================================================================
+ALTER INDEX RENAME
+================================================================================
+
+ALTER INDEX IF EXISTS myindex RENAME TO my_index
+
+--------------------------------------------------------------------------------
+
+(program
+  (statement
+    (alter_index
+      (keyword_alter)
+      (keyword_index)
+      (keyword_if)
+      (keyword_exists)
+      (identifier)
+      (rename_object
+        (keyword_rename)
+        (keyword_to)
+        (object_reference
+          (identifier))))))
+
+================================================================================
+ALTER INDEX
+================================================================================
+
+ALTER INDEX distributors SET (fillfactor = 75);
+ALTER INDEX distributors RESET (fillfactor);
+ALTER INDEX coord_idx ALTER COLUMN 3 SET STATISTICS 1000;
+ALTER INDEX distributors SET TABLESPACE fasttablespace;
```

Comment:
these individual checks are good but a quick "maximal" test setting as many options as possible would be useful for each of the complex object specifications here. Probably more manageable as individual cases than rolled in like these are

Reply:
I have separated them a little bit more, so that similar tests are together. I also added some more "complex" tests.